fbw-common/src/wasm/systems/systems/src/air_conditioning/cabin_pressure_controller.rs

```diff
@@ -255,17 +260,20 @@ impl<C: PressurizationConstants> CabinPressureController<C> {
         let (_, adirs_ambient_pressure) = self.adirs_values_calculation(adirs);
         let new_exterior_altitude: Length;
 
+        // If the ADIRUS are not aligned we take the ambient pressure from context
+        // This is to avoid the safety valve opening and the cabin delta pressure having to "adjust"
+        // after alignment. While they are not aligned we send No Computed Signal so the data is not displayed
+        // it's only used in internal calculations
         if !self.is_initialised {
-            self.exterior_pressure.reset(
-                adirs_ambient_pressure.unwrap_or_else(|| Pressure::new::<hectopascal>(Self::P_0)),
-            );
+            self.exterior_pressure
+                .reset(adirs_ambient_pressure.unwrap_or(context.ambient_pressure()));
             new_exterior_altitude =
                 self.calculate_altitude(self.exterior_pressure.output(), self.reference_pressure);
             self.is_initialised = true;
         } else {
             self.exterior_pressure.update(
                 context.delta(),
-                adirs_ambient_pressure.unwrap_or_else(|| Pressure::new::<hectopascal>(Self::P_0)),
+                adirs_ambient_pressure.unwrap_or(context.ambient_pressure()),
```

Comment:
How does the CPC get the exterior pressure? Does it have an independent pressure sensor? If not, it kind of defeats the point of using the ADR data output if we just end up magically getting the pressure anyway

Reply:
I added a comment to explain this in the code :) The CPC does not have an independent pressure sensor, we get the ambient pressure from context before ADIRS power up because:
1. The safety valve is a physical, mechanical valve that opens based on pressure differential, this pressure is not a "signal" but a physical pressure difference between the inside and the outside of the aircraft.
2. I have a low pass filter on the outside pressure to avoid delta P being too jerky when flying. Because of this, initialising external pressure to 1013 might result in you seeing delta P move when turning on the ADIRS, which you wouldn't in the real aircraft.

As the comment states, when the ADIRS are not powered the delta P signal has a No Ccomputed Data SSM, so any users of this value would not be able to get this info outside of the CPC.

Reply:
Hm, but why is this then the physical ambient pressure one time, and the ADR reported pressure the other time? That doesn't really make sense. If it's a physical quantity, it should always be the physical quantity. If it's data reported by the ADR, it should always be that (or a default if not valid). Maybe it would make sense to extract the safety valve into it's own component that always uses the physical quantity, and the CPC can then correctly always just use ADR data.

Reply:
Yeah I see what you mean and it's a good point, the safety valve should never use the ADR data. I'll change this. 👍

Reply:
As you suggested, I've separated the safety valve "signal" away from the CPC (and I've also done the same with the negative relief valve signal of the A380).

The CPC now only "sees" the ambient pressure from the ADIRS.

Thanks again for the suggestion!